### ISSUE TITLE
ios_acls state: parsed not working with version 1.3.0

### ISSUE THREAD

**Issue by ugarix** [ansible-collections/cisco.ios#242]
##### SUMMARY

State parsed is broken with 1.3.0 version. This is my task definition:

  - name: gather ACL configuration
    ios_acls:
      running_config: "{{ lookup('file', 'acl_to_parse.cfg') }}"
      state: parsed
    register: output2

With ios collection 1.3.0 I get this error:
```
TASK [gather ACL configuration] ****************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: jinja2.exceptions.UndefinedError: 'acl_name' is undefined
fatal: [ios1]: FAILED! => changed=false 
  module_stderr: |-
    Traceback (most recent call last):
      File "/home/vagrant/.ansible/tmp/ansible-local-4380vr3e36b3/ansible-tmp-1612803823.2115908-4416-10636061324202/AnsiballZ_ios_acls.py", line 102, in <module>
        _ansiballz_main()
      File "/home/vagrant/.ansible/tmp/ansible-local-4380vr3e36b3/ansible-tmp-1612803823.2115908-4416-10636061324202/AnsiballZ_ios_acls.py", line 94, in _ansiballz_main
        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
      File "/home/vagrant/.ansible/tmp/ansible-local-4380vr3e36b3/ansible-tmp-1612803823.2115908-4416-10636061324202/AnsiballZ_ios_acls.py", line 40, in invoke_module
        runpy.run_module(mod_name='ansible_collections.cisco.ios.plugins.modules.ios_acls', init_globals=None, run_name='__main__', alter_sys=True)
      File "/usr/local/lib/python3.8/runpy.py", line 207, in run_module
        return _run_module_code(code, init_globals, run_name, mod_spec)
      File "/usr/local/lib/python3.8/runpy.py", line 97, in _run_module_code
        _run_code(code, mod_globals, init_globals,
      File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
        exec(code, run_globals)
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_acls.py", line 1415, in <module>
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/cisco/ios/plugins/modules/ios_acls.py", line 1410, in main
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/config/acls/acls.py", line 43, in __init__
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/resource_module.py", line 46, in __init__
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/resource_module.py", line 64, in gather_current
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/resource_module.py", line 110, in get_facts
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/facts.py", line 121, in get_facts
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/facts/facts.py", line 131, in get_network_resources_facts
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/cisco/ios/plugins/module_utils/network/ios/facts/acls/acls.py", line 70, in populate_facts
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/network_template.py", line 63, in parse
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/network_template.py", line 38, in _deepformat
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/network_template.py", line 34, in _deepformat
      File "/tmp/ansible_ios_acls_payload_hhkjhpa6/ansible_ios_acls_payload.zip/ansible_collections/ansible/netcommon/plugins/module_utils/network/common/utils.py", line 740, in __call__
      File "/home/vagrant/ansible/env/lib/python3.8/site-packages/jinja2/environment.py", line 1090, in render
        self.environment.handle_exception()
      File "/home/vagrant/ansible/env/lib/python3.8/site-packages/jinja2/environment.py", line 832, in handle_exception
        reraise(*rewrite_traceback_stack(source=source))
      File "/home/vagrant/ansible/env/lib/python3.8/site-packages/jinja2/_compat.py", line 28, in reraise
        raise value.with_traceback(tb)
      File "<template>", line 1, in top-level template code
    jinja2.exceptions.UndefinedError: 'acl_name' is undefined
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1

```
I reverted back to 1.2.1 and it runs smoothly:
```
TASK [gather ACL configuration] ****************************************************************************************
ok: [ios1]

TASK [debug output] ****************************************************************************************************
ok: [ios1] => 
  msg:
    changed: false
    failed: false
    parsed:
    - acls:
      - aces:
        - destination:
            any: true
            port_protocol:
              eq: telnet
          dscp: af11"
          grant: deny
          protocol: tcp
          protocol_options:
            tcp:
              ack: true
          source:
            any: true
            port_protocol:
              eq: www
        name: R1_TRAFFIC
      afi: ipv6
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cisco.ios.ios_acls module, state: parsed

##### ANSIBLE VERSION

```paste below
(env) [vagrant@server1 ansible]$ ansible --version
ansible 2.10.5
  config file = /home/vagrant/ansible/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/vagrant/ansible/env/lib/python3.8/site-packages/ansible
  executable location = /home/vagrant/ansible/env/bin/ansible
  python version = 3.8.7 (default, Dec 27 2020, 09:01:11) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

##### CONFIGURATION

```paste below
DEFAULT_HOST_LIST(/home/vagrant/ansible/ansible.cfg) = ['/home/vagrant/ansible/hosts']
DEFAULT_STDOUT_CALLBACK(/home/vagrant/ansible/ansible.cfg) = yaml
HOST_KEY_CHECKING(/home/vagrant/ansible/ansible.cfg) = False
INTERPRETER_PYTHON(/home/vagrant/ansible/ansible.cfg) = /home/vagrant/ansible/env/bin/python3.8
```

##### OS / ENVIRONMENT

Centos 7 controller. Network device firmware irrevelant as state:parsed doesn't connect to devices.

##### STEPS TO REPRODUCE

RUn  the exact same task with cisco.ios collection 1.3.0

```yaml
- hosts: ios1
  gather_facts: no
  collections:
  - cisco.ios
  tasks:
  - name: Parse the provided ACLs configuration
    cisco.ios.ios_interfaces:
      running_config:
         "interface GigabitEthernet3"
      state: parsed
    register: output
  - name: debug output
    debug:
      msg: "{{ output }}"
  - name: gather ACL configuration
    ios_acls:
      running_config: "{{ lookup('file', 'acl_to_parse.cfg') }}"
      state: parsed
    register: output2
  - name: debug output
    debug:
      msg: "{{ output2 }}"
```

##### ACTUAL RESULTS

```paste below

```

**Comment by justjais** [comment]
@ugarix can you plz share the `acl_to_parse.cfg` config to triage the issue.

**Comment by ugarix** [comment]
(env) [vagrant@server1 ansible]$ more acl_to_parse.cfg 
"ipv6 access-list R1_TRAFFIC
deny tcp any eq www any eq telnet ack dscp af11"

**Comment by ugarix** [comment]
Providing the acl inline is no different as well:

  - name: Parse the provided ACLs configuration
    cisco.ios.ios_acls:
      running_config:
         "ipv6 access-list R1_TRAFFIC
         deny tcp any eq www any eq telnet ack dscp af11"
      state: parsed
    register: output1

**Comment by justjais** [comment]
@ugarix Thanks for sharing the `parsed.cfg` content and the failure was happening as we now check for `IPv6` instead of `ipv6`, how it comes under `sh access-list` output and parsed input cfg is expected to be the same as respective cmd out, though there was an issue with ios_acls parsed state example under module doc which will be fixed by #244 PR.

Once the PR is merged it'd resolve the logged issue, so closing the issue.
